frontend/src/api/bookmarks.ts

```diff
@@ -0,0 +1,107 @@
+import { API_V1_BASE_PREFIX } from '@/api/common'
+import { UsersClient } from '@/api/users'
+
+/**
+ * Access to the bookmarks part of the API.
+ */
+export class BookmarksClient {
+  private apiBaseUrl: string
+  private csrfToken: string | null
+  private currentUserId: string | null
+
+  constructor(apiBaseUrl?: string, csrfToken?: string) {
+    this.apiBaseUrl = apiBaseUrl ?? API_V1_BASE_PREFIX
+    this.csrfToken = csrfToken ?? null
+    this.currentUserId = null
+  }
+
+  /**
+   * Obtains the currently logged in user's information.
+   */
+  async fetchCurrentUser(): Promise<any> {
+    const userClient = new UsersClient()
+    const response = await userClient.fetchCurrentUserProfile()
+    this.currentUserId = response.id
+  }
+
+  /**
+   * Obtains the currently logged in user's bookmarks.
+   *
+   * @returns bookmarks list for the current user
+   */
+  async fetchBookmarks(): Promise<any> {
+    // Obtain the current user's information
+    await this.fetchCurrentUser()
+    const url = `${this.apiBaseUrl}bookmarks/list?user_id=${this.currentUserId}`
```

Comment:
The backend knows the current user and should only allow access to the current user's bookmarks - or any other entity.